src/Source/DoctrineDBALConnectionSourceIterator.php

```diff
@@ -79,7 +79,14 @@ public function valid(): bool
     public function rewind(): void
     {
         $statement = $this->connection->prepare($this->query);
-        $this->result = $statement->execute($this->parameters);
+
+        if (method_exists($statement, 'executeQuery')) {
```

Comment:
Can you add a comment about when dropping the check ?

Reply:
Done